fixitfelix/fix.py

```diff
@@ -107,20 +106,31 @@ def write_chunks_to_file(
     index_ranges: Chunk Indices that point to valid data slices
     group: TDMS Group inside the old tdms file
     channel: TDMS Channel inside group
-    cached_read: Determines which read method is used
+    segment_size: Sets size of each segment written to a TDMS file
     """
 
-    if cached_read:
-        channel_data = channel.read_data()
-        for (offset, length) in tqdm.tqdm(index_ranges):
-            data = channel_data[offset : offset + length]
-            new_channel = nptdms.ChannelObject(group.name, channel.name, data)
-            tdms_writer.write_segment([new_channel])
-    else:
-        for (offset, length) in tqdm.tqdm(index_ranges):
-            data = channel.read_data(offset=offset, length=length)
-            new_channel = nptdms.ChannelObject(group.name, channel.name, data)
+    clean_data = []
+    clean_data_nbytes = 0
+    for (offset, length) in tqdm.tqdm(index_ranges):
+        data = channel.read_data(offset=offset, length=length)
+        clean_data.append(data)
+        clean_data_nbytes += data.nbytes
+        # When segment_size is reached, a new segment is written to file
+        if clean_data_nbytes > segment_size * 1000000000:
```

Comment:
Use a named constant for `1000000000` ?

Reply:
You can also write `1_000_000_000` so it's less error prone.

Reply:
I think i am gonna go with @ccauet 's suggestion, because the value is just there to convert the segment size from gigabyte to byte. So it will always be a billion.